packages/common/src/index.ts

```diff
@@ -519,13 +498,8 @@ export default class Common extends EventEmitter {
    * @param hardfork Hardfork given to function as a parameter
    * @returns Hardfork chosen to be used
    */
-  _chooseHardfork(hardfork?: string | Hardfork | null, onlySupported: boolean = true): string {
-    if (!hardfork) {
-      hardfork = this._hardfork
-    } else if (onlySupported && !this._isSupportedHardfork(hardfork)) {
-      throw new Error(`Hardfork ${hardfork} not set as supported in supportedHardforks`)
-    }
-    return hardfork
+  _chooseHardfork(hardfork?: string | Hardfork | null): string {
+    return hardfork ?? this._hardfork
```

Comment:
I guess I would be a fan of now completely remove this internal function and everywhere replace with this `hardfork ?? this._hardfork` switch (like we do with options on other places), seems more natural to me.

Reply:
Yeah I think that's the right call. PR updated